lib/semantics/tools.cc

```diff
@@ -264,6 +265,14 @@ bool ExprHasTypeCategory(const evaluate::GenericExprWrapper &expr,
   return dynamicType.has_value() && dynamicType->category == type;
 }
 
+bool ExprTypeKindIsDefault(
+    const evaluate::GenericExprWrapper &expr, const SemanticsContext &context) {
+  auto dynamicType{expr.v.GetType()};
+  return dynamicType.has_value() &&
+      dynamicType->kind ==
+      context.defaultKinds().GetDefaultKind(dynamicType->category);
```

Comment:
This code will crash the compiler if the expression has a derived type.  It would be better if the function returned false.

Reply:
Done.